### DISCUSSION TITLE
PostgreSQL 15.1 and PHP 8.1.x Upgrade

### DISCUSSION THREAD

**Discussion by johnnyq** [nextcloud/all-in-one#1396] [Ideas] [answered]
It looks like the first point release of PostgreSQL 15.1 came out last week and PHP 8.1.x has been out for sometime.
PostgreSQL 15.x is supposed to be a huge improvement for database performance along with PHP 8.1
Any Speed improvement will be a plus with Nextcloud.
Is it a possibility to include these in the next big update or at least PHP 8.1.x?

**Comment by Zoey2936** [comment]
PostgreSQL 15.x is currently not supported by nextcloud see: https://github.com/nextcloud/server/pull/34645
PHP 8.1 is supported by nextcloud, so it should be possible

**Comment by johnnyq** [comment]
Thanks for pointing that out @Zoey2936

**Comment by szaimen** [comment]
Regarding Postgres 15, that should be possible with NC25, but it will most likely be added earliest after the AIO upgrade to 25 is out. 

As for PHP 8.1, I am thinking about adding it earliest for the NC26 upgrade to be extra sure that everything works.

**Comment by Zoey2936** [comment]
I wanted to also ask on this decency updates:
- alpine v3.17.0
- clamav v1
- redis v7

**Reply by szaimen** [reply]
> * alpine v3.17.0

I'll wait for 3.17.1. Also not sure if SSE will work eith that due to openssl 3.0. if not we need to add a known workaroind.

> * clamav v1

I'll also wait for the next patch release of that.

> * redis v7

~~Should be doable with the next patch release as well.~~ Will be added after the next release that comes with nc25.

**Reply by szaimen** [reply]
I guess we could also check if we can upgrade the talk container to ubuntu jammy...

**Reply by Zoey2936** [reply]
Yes, I think this should work.
But I also think we need a better overview about major dependency updates.

**Comment by szaimen** [comment]
Redis v7 is now released with v4.2.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by szaimen** [comment]
I created https://github.com/nextcloud/all-in-one/issues/1782 for the php 8.1 update for the Nextcloud container

**Comment by Zoey2936** [comment]
So the following major upgrades are still to be done at some point (right @szaimen? Or is something missing?):
- Nextcloud container alpine 3.17
- Nextcloud container php8.1
- talk migration to alpine (maybe alpine 3.18) or upgrade to Ubuntu jammy
- Clam AV v1


Wide future;
- Mastercontainer php8.2
- Nextcloud php8.2

**Reply by szaimen** [reply]
> * Nextcloud container alpine 3.17
> * Nextcloud container php8.1

tracked in https://github.com/nextcloud/all-in-one/issues/1782

> * talk migration to alpine (maybe alpine 3.18) or upgrade to Ubuntu jammy

tracked in https://github.com/nextcloud/all-in-one/issues/205

> * Clam AV v1

Should automatically get opened as soon as the next patch or minor version comes out

> Wide future;
> 
> * Mastercontainer php8.2
> * Nextcloud php8.2

Yeah, but lets create a new issue or thread for this, although is not important for now, imo

**Reply by szaimen** [reply]
Clamav 1.0 is now released with v4.4.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Comment by szaimen** [comment, marked as answer]
Except php 8.2 is everything released now with v4.3.0 or scheduled in other issues.

**Reply by johnnyq** [reply]
Fantastic!

**Comment by Zoey2936** [comment]
To continue this, I want to ask what the plan is with the following (big) dep updates (even if some already have Issues, I want to mention them), I know not all images already have alpine v3.18 variants:
- alpine v3.18 (waiting for v3.18.1?) — watchtower, domaincheck, borg, apache
- imaginary, alpine 3.18 for the final and go part
- will the Nextcloud container be updates to alpine v3.17 or v3.18?
- will the Nextcloud container be updated next to php 8.1 or 8.2?
- php 8.2? mastercontainer
- alpine v3.18 mastercontainer
- talk container 4.6.2-r1-alpine (alpine 3.18)
- Elastic v8 (ubuntu/debian based?)
- apache/httpd alpine 3.18
- (postgres has no alpine verison pinning)
- (clamav has no alpine verison pinning)
- (redis has no alpine verison pinning)
- office excluded from this list (ubuntu/debian based?)
- copying of docker/caddy/watchtower/spread binaries has no alpine version pinning
- nats binary has no OS at all

**Reply by szaimen** [reply]
can you create a new thread for ths? Thanks!

**Reply by Zoey2936** [reply]
I think this should be all

**Reply by Zoey2936** [reply]
> can you create a new thread for ths? Thanks!

yes
